runtime/lua/vim/lsp/rpc.lua

```diff
@@ -317,10 +339,10 @@ end
 --- Sends a request to the LSP server and runs {callback} upon response.
 ---
 ---@param method (string) The invoked LSP method
----@param params (table|nil) Parameters for the invoked LSP method
----@param callback fun(err: lsp.ResponseError|nil, result: any) Callback to invoke
----@param notify_reply_callback (function|nil) Callback to invoke as soon as a request is no longer pending
----@return boolean success, integer|nil request_id true, request_id if request could be sent, `false` if not
+---@param params (table?) Parameters for the invoked LSP method
```

Comment:
```suggestion
---@param params (table|nil) Parameters for the invoked LSP method
```

And for other many places: see https://github.com/neovim/neovim/blob/master/CONTRIBUTING.md#lua-docstrings

Reply:
I thought that using this syntax [for optional values](https://github.com/LuaLS/lua-language-server/wiki/Annotations#param) was preffered over the old `|nil`. Is it not?

Reply:
Throughout neovim codebase, `table|nil` is used instead of `table?` But if you meant an optional parameter, it should be `@param params? table` (without nil).

Reply:
> Throughout neovim codebase, table|nil is used instead of table?

Remember that Neovim's codebase is vast and written over years. Guidance can change without wanting to add churn by rewriting all old code. Nevertheless, new(ly touched) code should follow current guidance (`:h develop`; I don't think we've explicitly formalized this specific aspect -- @lewis6991 @mfussenegger @gpanders @MariaSolOs ?).